rfcs/rfc-team-notifications.md

```diff
@@ -0,0 +1,256 @@
+
+# Detailed design
+
+A notification informs someone about something that has happened or will happen (eg, a comment was added to an issue, or a meeting has been scheduled).
+
+A notification can take various forms, and depends on the user's environment (eg, desktop vs mobile, Mac vs PC, etc). It can be an app notification or a desktop / system notification. It may just display an icon or count, or it may display an alert or play a sound. Note that some notifications are easier to ignore than others. This is amplified below in the [Current process](#current-process) section.
+
+Sometimes, a team member needs to be "heads down" on a task and prefers to be uninterrupted for a time. It is not expected that this person would re-configure their notifications, although they may temporarily mute them or close the offending app or just wilfully ignore them. This is an edge case and can be managed by advance notice that this person will be less responsive than usual, and handled accordingly.
+
+Sometimes, a team member is not 100% allocated to one team and will have time / days where they are unavailable. From their point of view, they will need to manage incoming notifications so that they can remain focused on their current team while, ideally, remaining responsive to their other team(s) -- which is the theme of this RFC. From others' point of view, notice of their unavailability would be helpful to avoid any blockers -- asking questions ahead of time, or working on something else while awaiting an answer. This is an edge case similar to the one above.
+
+Another edge case is notifications during off-work hours. It is not generally expected that team members be responsive when away from work, so this is out of scope.
+
+The priority / level of a notification needs to be considered -- some should be seen as soon as possible while others can maybe wait a day.
+
+Finally is the subject of responding to notifications -- some should be acknowledged (eg, thumbs up emoji), some may need a response (eg, meeting invite), and some don't need a response. This is obviously contextual and guidance is provided in the [Best practices](#best-practices) section below.
+
+## Current process
+
+Team members use the following systems daily:
+
+### Microsoft Teams
+
+This software is primarily used for individual / group / team messaging and conferencing. MS Teams is the main team collaboration application used by BCROS teams.
+
+MS Teams supports "activity" and "banner" notifications for the Activity page and each team (aka channel) the user has access to. In addition to bolded headers in the app and a badge on the tray icon, Teams may generate popups and a sound through the desktop (eg, Windows Notifications), which are hard to ignore.
+
+In the author's experience, MS Teams generates a lot notifications, by virtue of the many teams and chat groups subscribed to.
+
```

Comment:
Take in consideration Teams messages retention policies as messages are purged after a while. Distribution of information that is meant for future reference should be done via e-mails or other methods

Reply:
I have added the following (in my next commit):

> A consideration is that MS Teams messages are purged after a while according to retention policies. Hence, distribution of information that is meant for future reference should be done by email (where messages can be filed into folders permanently) or other methods.

Reply:
Some contractors use the browser version of MS Teams. In this case, it is recommended to ensure that the tab is exempted from the browser's memory saving configuration and to also have the permission to show desktop notification.